client/src/App.jsx

```diff
@@ -1,17 +1,69 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import Reviews from './components/R&R/Reviews.jsx';
 import Questions from './components/Q&A/questions.jsx';
-
+import axios from 'axios';
+import { ProductContext } from './context';
 import ProductOverview from 'components/overview/ProductOverview.jsx';
+import { composeParams } from './utils'
+
+const port = 4000;
+const baseUrl = `http://localhost:${port}`;
+
 
 const App = () => {
-  return (
-    <div>
-      <ProductOverview />
-      <Reviews />
-      <Questions/>
-    </div>
-  );
+  const [reviews, setReviews] = useState(null);
+  const [reviewsMeta, setReviewsMeta] = useState(null);
+  const [product, setProduct] = useState(null);
+  const [productStyles, setProductStyles] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [productId, setProductId] = useState(18082);
+  const [error, setError] = useState('');
+
+  function fetchData(route, setData, params) {
+    return new Promise((resolve, reject) => {
+      const requestURL = params ? `${baseUrl}${route}${composeParams(params)}` : `${baseUrl}${route}`
+      axios.get(requestURL)
+        .then((results) => {
+          setData(results.data);
```

Comment:
This could potentially break if setting state doesn't complete before the promise resolves.